src/main/java/org/candlepin/model/DistributorVersionCurator.java

```diff
@@ -33,66 +32,137 @@ public DistributorVersionCurator() {
+    /**
+     * Retrieves a list of {@link DistributorVersion}s that has a name that equals or has a substring of
+     * the provided name.
+     *
+     * @param name
+     *  the name of the {@link DistributorVersion}s to retrieve
+     *
+     * @return a list of distributor versions that has a name equal to or has a substring of the provided
+     *  name
+     */
     public List<DistributorVersion> findByNameSearch(String name) {
-        return currentSession()
-            .createCriteria(DistributorVersion.class)
-            .add(Restrictions.like("name", "%" + name + "%")).list();
+        if (name == null) {
+            return new ArrayList<>();
+        }
+
+        String query = "SELECT d FROM DistributorVersion d WHERE d.name LIKE :name";
+
+        return this.currentSession().createQuery(query, DistributorVersion.class)
+            .setParameter("name", "%" + name + "%")
+            .getResultList();
     }
 
-    @SuppressWarnings("unchecked")
+    /**
+     * Retrieves a list of {@link DistributorVersion} that has a {@link DistributorVersionCapability} with a
+     * name that equals the provided capability name.
+     *
+     * @param capability
+     *  the name of the {@link DistributorVersionCapability} to retrieve {@link DistributorVersion}s for
+     *
+     * @return a list of distributor versions that have a capability that matches the provided name
+     */
     public List<DistributorVersion> findByCapability(String capability) {
-        List<DistributorVersionCapability> caps = currentSession()
-            .createCriteria(DistributorVersionCapability.class)
-            .add(Restrictions.eq("name", capability)).list();
+        if (capability == null || capability.isBlank()) {
+            return new ArrayList<>();
+        }
 
-        List<DistributorVersion> distVers = new ArrayList<>();
+        String query = "SELECT d FROM DistributorVersionCapability d WHERE d.name = :name";
 
-        for (DistributorVersionCapability dvc : caps) {
-            distVers.add(dvc.getDistributorVersion());
+        List<DistributorVersionCapability> distributorVersionCapabilities = this.currentSession()
+            .createQuery(query, DistributorVersionCapability.class)
+            .setParameter("name", capability)
+            .getResultList();
+
+        List<DistributorVersion> result = new ArrayList<>();
+        for (DistributorVersionCapability dvc : distributorVersionCapabilities) {
+            result.add(dvc.getDistributorVersion());
         }
 
-        return distVers;
+        return result;
     }
 
-    @SuppressWarnings("unchecked")
+    /**
+     * Retrieves a {@link DistributorVersion} that has an ID equal to the provided ID.
+     *
+     * @param id
+     *  the ID to retrieve a distributor version for
+     *
+     * @return a distributor version that has an ID equal to the provided ID, or null if one does not exist
+     */
     public DistributorVersion findById(String id) {
```

Comment:
Same deal with this method: the universal `get(id)` from AHC can replace this directly.

Reply:
Removed